### PR TITLE
feat(behavior_velocity_planner): hold stop margin distance in stop line module

### PR DESCRIPTION
**Pull request by satoshi-ota**
Signed-off-by: satoshi-ota <satoshi.ota928@gmail.com>

## Description

This PR is resubmission of #1433. (#1433 has been reverted in #1645)

Please see https://github.com/autowarefoundation/autoware.universe/pull/1433 description.

test perform (`stop_duration_sec`= 5.0[s], `hold_stop_margin_distance`= 2.0[m])

https://user-images.githubusercontent.com/44889564/185872691-8702b22c-2674-467f-9ba7-253809cf2ec3.mp4

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by taikitanaka3** [comment]
LGTM（this will not affect duplicate route process）

**Review by taikitanaka3 — APPROVED** [review]
LGTM

**Comment by satoshi-ota** [comment]
Resolve conflicts.

File: planning/behavior_velocity_planner/src/scene_module/stop_line/scene.cpp [hunk 1]
```diff
@@ -81,46 +81,88 @@ bool StopLineModule::modifyPathVelocity(
...
+        path->points, planner_data_->current_pose.pose.position, 0.0);
+
+      if (!stopped_pose) {
+        break;
+      }
+
+      SegmentIndexWithPose ego_pos_on_path;
+      ego_pos_on_path.pose = stopped_pose.get();
+      ego_pos_on_path.index = motion_utils::findFirstNearestSegmentIndexWithSoftConstraints(
+        path->points, stopped_pose.get());
```
Comment (takayuki5168) [inline planning/behavior_velocity_planner/src/scene_module/stop_line/scene.cpp:131]: use findEgoSegmentIndex which is a wrapper function of findFirstNearestSegmentIndexWithSoftConstraints with proper praramters.

Comment (satoshi-ota) [reply planning/behavior_velocity_planner/src/scene_module/stop_line/scene.cpp]: Fix in https://github.com/autowarefoundation/autoware.universe/pull/1648/commits/0527bba1710ce7d7a8bbd6c1d55da6268b42357c

**Review by takayuki5168 — APPROVED** [review]
LGTM

**Pull request merged**